modes/segmentation/src/segmentationButtons.ts

```diff
@@ -0,0 +1,222 @@
+import type { Button } from '@ohif/core/types';
+
+function _createSetToolActiveCommands(toolName) {
+  return [
+    {
+      commandName: 'setToolActive',
+      commandOptions: {
+        toolName,
+      },
+      context: 'CORNERSTONE',
+    },
+  ];
+}
+
+const executeSetToolActive = (commandsManager, toolName) => {
+  commandsManager.runCommand('setToolActive', { toolName }, 'CORNERSTONE');
+};
+
+const executeSetBrushProperty = (commandsManager, property, toolNames, value) => {
+  commandsManager.runCommand(
+    property,
+    {
+      value,
+      toolNames,
+    },
+    'SEGMENTATION'
+  );
+};
+
+const executeSetBrushSize = (commandsManager, value) => {
+  executeSetBrushProperty(commandsManager, 'setBrushSize', ['CircularBrush', 'SphereBrush'], value);
+};
+
+const executeSetEraserSize = (commandsManager, value) => {
+  executeSetBrushProperty(
+    commandsManager,
+    'setBrushSize',
+    ['CircularEraser', 'SphereEraser'],
+    value
+  );
+};
+
+const executeSetThresholdSize = (commandsManager, value) => {
+  executeSetBrushProperty(
+    commandsManager,
+    'setBrushSize',
+    ['ThresholdCircularBrush', 'ThresholdSphereBrush'],
+    value
+  );
+};
+
+const executeSetThresholdRange = (commandsManager, values) => {
+  commandsManager.runCommand(
+    'setThresholdRange',
+    { values, toolNames: ['ThresholdCircularBrush', 'ThresholdSphereBrush'] },
+    'SEGMENTATION'
+  );
+};
+
+const toolbarButtons: Button[] = [
+  {
+    id: 'BrushTools',
+    uiType: 'ohif.buttonGroup',
+    props: {
+      groupId: 'BrushTools',
+      items: [
+        {
+          id: 'Brush',
+          icon: 'icon-tool-brush',
+          label: 'Brush',
+          evaluate: {
+            name: 'evaluate.cornerstone.segmentation',
+            options: { toolNames: ['CircularBrush', 'SphereBrush'] },
+          },
+          commands: _createSetToolActiveCommands('CircularBrush'),
+          options: [
+            {
+              name: 'Radius (mm)',
+              id: 'brush-radius',
+              type: 'range',
+              min: 0.5,
+              max: 99.5,
+              step: 0.5,
+              value: 25,
+              onChange: executeSetBrushSize,
```

Comment:
Is there any reason the value isn't simply included as 'value' in the call and then you can just execute the commands with the default properties, rather than needing an onChange?  I like having the onChange, I just don't think this particular use case is one that is needed.

Reply:
The issue is that we start adding executable code definitions to the toolbar definitions - which IMHO should be 99% declarative.  Yes, once in a while it needs an onChange listener, but almost entirely I think it should be defined like this:
```
 {
              name: 'Radius (mm)',
              id: 'brush-radius',
              type: 'range',
              min: 0.5,
              max: 99.5,
              step: 0.5,
              value: 25,
              // onChange: executeSetBrushSize,
              commands: { commandName: 'setBrushRadius', commandOptions: {  tools: ['CircularBrush', 'SphereBrush'] }},
            },
```

So, instead of the onChange, there is a commands inside the option, and that commands will get executed with the run option and will include the max, step etc, PLUS the actual current value.  That way it is entirely declarative and all the actual logic is inside the commands module 'setBrushRadius'
Basically - the onChange should be a commands list so we have 1 pattern for how things are done rather than 1 listener pattern, and one commands pattern.